src/Search/Services/IndexableService.php

```diff
@@ -2,29 +2,53 @@
 
 namespace SilverStripe\FullTextSearch\Search\Services;
 
+use SilverStripe\Core\Config\Config;
+use SilverStripe\Core\Config\Configurable;
 use SilverStripe\Core\Extensible;
 use SilverStripe\Core\Injector\Injectable;
+use SilverStripe\FullTextSearch\Search\FullTextSearch;
 use SilverStripe\ORM\DataObject;
-use SilverStripe\ORM\Tests\MySQLDatabaseTest\Data;
+use SilverStripe\Security\Member;
 
 /**
- * Checks if a DataObject is publically viewable thus able to be added or retrieved from a publically searchable index
- * Caching results because these checks may be done multiple times as there a few different code paths that search
- * results might follow in real-world search implementations
+ * Checks if a DataObject is publicly viewable, thus able to be added to or retrieved from a publicly searchable index.
+ * Results are cached because these checks may be run multiple times, as there a few different code paths that search
+ * results might follow in real-world search implementations.
  */
 class IndexableService
 {
-
     use Injectable;
     use Extensible;
+    use Configurable;
+
+    /**
+     * Skip the canView() check at a class level to increase performance of solr_redindex and retrieving search results.
+     * Be careful as this may lead to content showing in search results that should not be there such as non-public,
+     * cms-user-only content.  This may potentially happen via edge cases such as skipping checks where subclasses
+     * are involved.
+     *
+     * @var array namespaced classes to skip canView() check.
+     */
+    private static $classes_that_skip_canview_check = [];
```

Comment:
This configuration was originally targeted specifically at the inlet stage, and I think it needs to change back, or at least be split in two (inlet and outlet).

There are perfectly valid use-cases for canView checks to operate only at the outlet stage - a primary example would be reintroducing the ability to search in the draft stage for logged in users. If this config is applied to both stages it will make it impossible to restore sites to the behaviour they had before upgrading to 3.7.

This will require adjustments to the `isIndexable` method to add a parameter for whether the caller is at the inlet/outlet stage, or a separate method (`isQueryable`?), to separate the behaviour.

Reply:
yup good point, will update